### DISCUSSION TITLE
[QUESTION] Is it possible to declaratively create a Volume and PV with ReadWriteMany and a larger size than my PVCs?

### DISCUSSION THREAD

**Discussion by jsolbrig** [longhorn/longhorn#4555] [Q&A]
I have a use case where I would like to create a Volume and PV with ReadWriteMany and give it a size large enough to accommodate many PVCs. 

Currently, I know how to:
- Create a large volume via the UI, then create PVCs that use it.
- Create a PVC with a resource request and allow new PVs and Volumes to be dynamically provisioned.

Instead, I would like to declaratively create a large (e.g. 100GB) Volume, then create smaller (e.g. 100MB) PVCs to use it. Is this possible?

If that isn't possible, is there a way for me to create a Volume by sending a request to an API endpoint in the Longhorn UI?

**Comment by joshimoo** [comment]
Sounds like you want to share a PV with many different PVC's this isn't a supported operation in Kubernetes since PV <> PVC are 1 to 1 mappings.

Below is a really old example from v1.0.X where we added an example of a NFS provisioner, that utilizes a single longhorn PV,PVC to then provision pvc -> pv on top of the longhorn volume. https://longhorn.io/docs/archives/1.0.2/advanced-resources/rwx-workloads/ since this is really old and we now have native RWX integration this isn't really supported anymore.

**The important part is that there is probably a better approach to accomplish your use case**
Feel free to explain your use case and we will provide better guidance on how to accomplish it :)

**Comment by maxweisspoker** [comment]
I wanted to bump this thread to comment how CephFS can be hacked into being a shared volume. [Their instructions](https://rook.io/docs/rook/v1.10/Storage-Configuration/Shared-Filesystem-CephFS/filesystem-storage/#shared-volume-creation) basically have you create a second PV yaml object that has the same data as the original PV, in order to point the PV object to use the same underlying data for the volume.  I was wondering if something like that is possible for Longhorn's non-NFS volume. I know the answer is probably not given that Longhorn's volume replication happens as a master-with-eventual-consistency-to-the-others type of replication, but in the event that something like this is possible, I wanted to bring it up. I would prefer not to use NFS, and the only reason I use rook-ceph over Longhorn is the ability to share CephFS volumes in this faux-rwx manner.
